File: Python-Basics/moving.py
free_width = int(input())
free_length = int(input())
free_height = int(input())

free_space_available = free_height * free_length * free_width
is_filled = True
command = input()
while command != "Done":
    current_space = int(command)
    free_space_available -= current_space

    if free_space_available <= 0:
        is_filled = False
        break
    command = input()


if is_filled:
    print(f"{abs(free_space_available)} Cubic meters left.")
else:
    print(f"No more free space! You need {abs(free_space_available)} Cubic meters more.")

File: JS-Basics/While Loop Exercise/moving.js
function calc(input){
    let index = 0;
    let spaceWidth = Number(input[index]);
    index++;
    let spaceLength = Number(input[index]);
    index++;
    let spaceHeight = Number(input[index]);
    index++;
    let totalSpace = spaceHeight * spaceLength * spaceWidth;
    let command = input[index];
    let isSuccessful = true;
    let sumSpace = 0;
    while(command !== "Done"){
        let currentBox = Number(command);
        sumSpace += currentBox
        if(sumSpace >= totalSpace){
            console.log(`No more free space! You need ${Math.abs(sumSpace - totalSpace)} Cubic meters more.`);
            isSuccessful = false;
            break;
        }
        index++;
        command = input[index];
    }
    if(isSuccessful){
        console.log(`${Math.abs(sumSpace - totalSpace)} Cubic meters left.`)
    }
}
calc(["10", 
"1",
"2",
"4", 
"6",
"Done"])


;
